### DISCUSSION TITLE
How to trigger from Nodejs Pusher to laravel Echo

### DISCUSSION THREAD

**Discussion by drpshtiwan** [soketi/pws#740] [Q&A]
How can I trigger any event to Laravel echo with the Nodejs or pusher PHP package. This is my configuration for Nodejs.
```javascript
const Pusher = require("pusher")

const pusher = new Pusher({
    appId: "app-id",
    key: "app-key",
    secret: "app-secret",
    useTLS: false, 
    host: "127.0.0.1",
    port: 6001, 
  });

pusher.trigger('websocket','WebsocketEvent',{message:'FROM NODE'})
```

This is the Laravel echo configuration:
```javascript
import Echo from 'laravel-echo';

import Pusher from 'pusher-js';
window.Pusher = Pusher;

window.Echo = new Echo({
    broadcaster: 'pusher',
    key: import.meta.env.VITE_PUSHER_APP_KEY, //app-key
    wsHost: import.meta.env.VITE_PUSHER_HOST, // 127.0.0.1
    wsPort: import.meta.env.VITE_PUSHER_PORT , // 6001
    wssPort: import.meta.env.VITE_PUSHER_PORT, // 6001
    forceTLS: false ,
    disableStats: true,
    enabledTransports: ['ws', 'wss'],
});

this.addEventListener('load',() => {
    Echo.channel('websocket').listen('WebsocketEvent', (e) => {
        console.log(e)
    })
})
```

Is there any way to trigger events to Laravel Echo?

**Comment by rennokki** [comment]
Make sure to read the [Pusher Node.js SDK documentation](https://github.com/pusher/pusher-http-node)

**Reply by drpshtiwan** [reply]
I read the documentation but it did not make sense. Can you tell what did I do wrong?

**Reply by ssatz** [reply]
I tried, but it didn't work.
Let me know know if you able to find the solution

**Comment by vivekkotadiya** [comment]
@rennokki I am new to Soketi and I want to integrate it on one site where we are used Pusher. But due to limit in concorrent users and messages we are planning to migrate to Soketi. Currently, we are building app in Node Js and Next Js (Frotn-end) so need integration guide here. If we can use Soketi with Pusher how can be manage unlimited concorrent connections and messages. Because I tried to Pusher package for Node Js ingration and when I run it's trigger even it counts messages and connections in Pusher dashboard. So I think it will restrict messages and connection based on active. Also, in soketi app config which app id, secret ket etc should be used?

Can you guide me for that how I can integrate it with node js for triggering events from soketi using Pusher? Also, how I can handle concurrent connection with horizontal scaling. I am using cloudways hosting as of now which is use Digital Ocean droplet.

Here is my demo code sample.
- https://prnt.sc/r1ODgqFw85Ev
- https://prnt.sc/1MP3YUIoNsW-

I tested my connection with soketi from postman and it is showing connected.
- https://prnt.sc/psF-_EP87tfG
